File: backend/local-api-module/src/main/java/org/vyatsu/localApiModule/entity/purchase/PurchaseItem.java
package org.vyatsu.localApiModule.entity.purchase;

import jakarta.persistence.*;
import lombok.*;
import org.vyatsu.localApiModule.entity.item.Item;

import java.util.LinkedHashSet;
import java.util.Set;

@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
@Builder
@Entity
@Table(name = "purchase_items")
public class PurchaseItem {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id", nullable = false)
    private Long id;

    @Column(name = "quantity", nullable = false)
    private Double quantity;

    @ManyToOne
    @JoinColumn(name = "id_purchase", nullable = false)
    private Purchase purchase;

    @ManyToOne
    @JoinColumn(name = "id_item", nullable = false)
    private Item item;

    @OneToMany(mappedBy = "purchaseItem")
    private Set<ContractorPurchaseItem> contractorPurchaseItems = new LinkedHashSet<>();
}

File: app/src/main/java/com/bytebusters/android/mpsystem/entity/purchase/PurchaseItem.kt
package com.bytebusters.android.mpsystem.entity.purchase

import com.bytebusters.android.mpsystem.entity.item.Item
import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.GeneratedValue
import jakarta.persistence.GenerationType
import jakarta.persistence.Id
import jakarta.persistence.JoinColumn
import jakarta.persistence.ManyToOne
import jakarta.persistence.Table
import lombok.AllArgsConstructor
import lombok.Data
import lombok.Getter
import lombok.NoArgsConstructor
import lombok.Setter

@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
@Data
@Entity
@Table(name = "purchase_items")
class PurchaseItem {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id", nullable = false)
    private val id: Long? = null

    @Column(name = "count", nullable = false)
    private val count = 0

    @ManyToOne
    @JoinColumn(name = "id_purchase", nullable = false)
    private val purchase: Purchase? = null

    @ManyToOne
    @JoinColumn(name = "id_item", nullable = false)
    private val item: Item? = null
}
